### PR TITLE
Hatch subsystem

### PR DESCRIPTION
**Pull request by ycyao216**
Implementing hatch subsystem.

### CODE REVIEW

**Review by rmeno12 — APPROVED** [review]
Everything looks good!

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
   /* Setup each subsystem */
   public static DriveTrain m_drive = new DriveTrain();
-
+  public static Hatch m_hatchIntake = new Hatch();
```
Comment (rmeno12) [inline src/main/java/frc/robot/Robot.java]: the naming is slightly odd here. i suggest either changing the variable to m_hatch or the subsystem to HatchIntake

```suggestion
  public static Hatch m_hatch= new Hatch();
```

File: src/main/java/frc/robot/subsystems/Hatch.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   * being used
+   */
+
+  public Hatch() {
+    this.m_launcher = new DoubleSolenoid(RobotMap.m_pcm12v, 0, 1);
+    this.m_beak = new DoubleSolenoid(RobotMap.m_pcm12v, 2, 3);
+  }
+
+  // simple solenoid operation: push out and turn off immediately
+  public void pushPneumaticsAutoRetract(DoubleSolenoid solenoid){
```
Comment (pvu-github) [inline src/main/java/frc/robot/subsystems/Hatch.java]: For consistency, rename to be "pneumaticsExtendAutoRetract()"

File: src/main/java/frc/robot/subsystems/Hatch.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  public void launchHatch(){
+    pushPneumaticsAutoRetract(this.m_launcher);
+  }
+
+  @Override
+  public void initDefaultCommand() {
+    // Set the default command for a subsystem here.
+    // setDefaultCommand(new MySpecialCommand());
```
Comment (pvu-github) [inline src/main/java/frc/robot/subsystems/Hatch.java]: We want to have the beak extended at the beginning of the match.

**Review by pvu-github — APPROVED** [review]
Awesome work! Please address the 2 comments I made and merge your code using the "Merge pull request" button.

Make sure it compiles first!

**Pull request merged**